library/core/class.format.php

```diff
@@ -955,6 +958,35 @@ protected static function isHtml($Text) {
     }
 
     /**
+     * Checks to see if $html has a spoiler element and replaces the element.
+     *
+     * @param string $html A html-formatted string
+     * @return string
+     */
+    protected static function replaceSpoilers($html) {
+        // Check if $html contains spoiler classes before we do the heavy lifting.
+        $found = false;
+        foreach(Gdn_Format::$spoilerClasses as $spoiler) {
+            if (strpos($html, $spoiler)) {
+                $found = true;
+            }
+        }
+        if (!$found) {
+            return $html;
+        }
+
+        // Transform $html into a dom object and replace the spoiler block.
+        if (!defined('HDOM_TYPE_ELEMENT')) {
+            require_once(PATH_LIBRARY . '/vendors/simplehtmldom/simple_html_dom.php');
+        }
+        $html = str_get_html($html);
+        foreach (Gdn_Format::$spoilerClasses as $spoiler) {
+            $html->find('.' . $spoiler, 0)->outertext = t('Spoiler Replacement', t('Spoiler'));
```

Comment:
Check to see if `$html->find('.', implode(',.'.self::$spoilerClasses))` works instead of the loop.

Also try and make the translation code work on its own. So something like `t('(Spoiler)')`.